x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/store/NativeRolesStore.java

```diff
@@ -529,9 +537,63 @@ public void onFailure(Exception e) {
         }
     }
 
+    public void ensureBuiltinRolesAreQueriable(ActionListener<Void> listener) {
+        if (enabled == false) {
+            listener.onFailure(new IllegalStateException("Native role management is disabled"));
+            return;
+        }
+        final SecurityIndexManager frozenSecurityIndex = securityIndex.defensiveCopy();
+        if (frozenSecurityIndex.indexExists() == false) {
+            logger.debug("security index does not exist");
+            listener.onResponse(null);
+        } else if (frozenSecurityIndex.isAvailable(SEARCH_SHARDS) == false) {
+            listener.onFailure(frozenSecurityIndex.getUnavailableReason(SEARCH_SHARDS));
+        } else if (frozenSecurityIndex.areReservedRolesIndexed()) {
+            logger.debug("security index already contains the latest reserved roles indexed");
+            listener.onResponse(null);
+        } else {
+            indexReservedRoles(
+                frozenSecurityIndex,
+                ActionListener.wrap(onResponse -> frozenSecurityIndex.markReservedRolesAsIndexed(listener), listener::onFailure)
+            );
+        }
+    }
```

Comment:
Check if the index metadata custom map indicates if the built-in reserved roles have already been indexed. If not, they are, and then the index metadata is marked as such.